starlite/contrib/repository/testing/generic_mock_repository.py

```diff
@@ -0,0 +1,192 @@
+"""A repository implementation for tests.
+
+Uses a `dict` for storage.
+"""
+from __future__ import annotations
+
+from datetime import datetime
+from typing import TYPE_CHECKING, Generic, TypeVar
+from uuid import uuid4
+
+from starlite.contrib.repository.abc import AbstractRepository
+from starlite.contrib.repository.exceptions import ConflictError, RepositoryError
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Hashable, Iterable, MutableMapping
+    from typing import Any
+
+    from starlite.contrib.repository.types import FilterTypes
+    from starlite.contrib.sqlalchemy import base
+
+ModelT = TypeVar("ModelT", bound="base.Base | base.AuditBase")
+MockRepoT = TypeVar("MockRepoT", bound="GenericMockRepository")
+
+
+class GenericMockRepository(AbstractRepository[ModelT], Generic[ModelT]):
+    """A repository implementation for tests.
+
+    Uses a `dict` for storage.
+    """
+
+    collection: MutableMapping[Hashable, ModelT]
+    model_type: type[ModelT]
+
+    def __init__(self, id_factory: Callable[[], Any] = uuid4, **_: Any) -> None:
+        super().__init__()
```

Comment:
Might as well propagate kwargs i think

Reply:
Well the idea here is to swallow the kwargs. 

If an instance of this mock repository gets monkey-patched over the top of a SQLAlchemy repository (to test business logic in a service object, for example), the code that calls it will pass a session to the constructor. If that were passed up to `AbstractRepository.__init__()`, which calls `super().__init__(**kwargs)` itself, it will throw a `TypeError`.

I think it is better for the mock repo to swallow unexpected kwargs than to move that logic into the base class, where unexpected kwargs would then also get swallowed at runtime.

Happy to discuss any of this ofc.